core/src/main/java/org/kiwix/kiwixmobile/core/main/TableDrawerAdapter.kt

```diff
@@ -0,0 +1,134 @@
+/*
+ * Kiwix Android
+ * Copyright (c) 2020 Kiwix <android.kiwix.org>
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package org.kiwix.kiwixmobile.core.main
+
+import android.graphics.Typeface
+import android.view.View
+import android.view.ViewGroup
+import androidx.recyclerview.widget.RecyclerView.Adapter
+import androidx.recyclerview.widget.RecyclerView.ViewHolder
+import kotlinx.android.synthetic.main.section_list.titleText
+import org.kiwix.kiwixmobile.core.R
+import org.kiwix.kiwixmobile.core.base.adapter.BaseViewHolder
+import org.kiwix.kiwixmobile.core.extensions.ViewGroupExtensions.inflate
+import java.lang.IllegalStateException
+
+class TableDrawerAdapter constructor(private val listener: TableClickListener) :
+  Adapter<ViewHolder>() {
+
+  private var title: String = ""
+  private var selectedPosition: Int = 0
+  private val sections: MutableList<DocumentSection> = mutableListOf()
+
+  fun setSections(sections: List<DocumentSection>) {
+    selectedPosition = 0
+    this.sections.clear()
+    this.sections.addAll(sections)
+  }
+
+  fun setTitle(title: String) {
+    this.title = title
+  }
+
+  override fun getItemViewType(position: Int): Int {
+    return if (position == 0) 0
+    else 1
+  }
+
+  override fun onCreateViewHolder(
+    parent: ViewGroup,
+    viewType: Int
+  ): ViewHolder {
+    val view = parent.inflate(R.layout.section_list, false)
+    if (viewType == 0) return HeaderTableDrawerViewHolder(view)
+    return SectionTableDrawerViewHolder(view)
+  }
+
+  override fun getItemCount(): Int = sections.size + 1
+
+  override fun onBindViewHolder(holder: ViewHolder, position: Int) {
+    when (holder) {
+      is HeaderTableDrawerViewHolder -> {
+        holder.bind(title)
+        holder.itemView.setOnClickListener { v: View ->
+          updateSelection(position)
+          listener.onHeaderClick(v)
+        }
+      }
+      is SectionTableDrawerViewHolder -> {
+        holder.bind(sections[position - 1])
+        holder.itemView.setOnClickListener { v: View? ->
+          updateSelection(position)
+          listener.onSectionClick(v, selectedPosition - 1)
```

Comment:
`selectedPosition - 1` reads like an error to me but I haven't manually tested this code, are you aware of how to test this code?

Reply:
>We are always subtracting 1 from the position because of the 0th position
which is reserved for the title. Also, I have tested the code manually by
running the app. Didn't found any crash or error.

right because `updateSelection` is just a roundabout way of saying `selectedPosition = position`

Definitely purging `selectedPosition` from this class will have benefits

Reply:
>And selectedPosition is just used as a
reference to notify the adapter of data change at a specific position,
instead of using notifyDataSetChanged.

What change is happening though? There is no visual alteration nor is the data in anyway changed because of a click. `selectedPosition` is utterly useless and by getting rid of it we can get rid of more code.

If you can please refrain from replying via email, it adds it as an independent comment making it harder to keep discussions contained